alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/endpoint/oauth/OAuthCustomEndpoint.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.descriptor.config.field.endpoint.oauth;
+
+import org.springframework.http.ResponseEntity;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.alert.common.action.CustomEndpointManager;
+import com.synopsys.integration.alert.common.descriptor.config.field.endpoint.ButtonCustomEndpoint;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.rest.HttpServletContentWrapper;
+import com.synopsys.integration.alert.common.rest.ResponseFactory;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+
+public abstract class OAuthCustomEndpoint extends ButtonCustomEndpoint {
```

Comment:
I think it might be best if we make this extend `CustomEndpoint` so that we aren't bound to return a `String` later. We should start moving away from `ResponseEntity<String>` IMO. If we do that, we can just return a `OAuthEndpointResponse` rather than wrapping and manually serializing with `Gson`.

Reply:
We will need to refactor the CustomEndpointManager and CustomEndpoint.  Although CustomEndpoint uses a generic R the method always returns a Response<String>